packages/frontend/src/utils/wallet.js

```diff
@@ -295,7 +295,7 @@ class Wallet {
         }
 
         const { data: recoveryMethods } = await this.getRecoveryMethods();
-        const methodsToRemove = recoveryMethods.filter((method) => method.kind !== 'ledger');
+        const methodsToRemove = (recoveryMethods || []).filter((method) => method.kind !== 'ledger');
```

Comment:
Is this related to your previous changes?

It looks like the destructure on L297 is broken - `this.getRecoveryMethods()` returns an array but it gets destructured like an object with a `data` property. The fallback you're adding will always be required so long as the wrong data type is returned.

The correct fix would appear to be to change L297 to:
```js
const recoveryMethods = await this.getRecoveryMethods();
```

If you want to make the changes to that line then I think it's fine either in this PR or a new one. But let's please remove this fallback and address the root of the issue instead. Once addressed, the fallback should never be required.

Reply:
Right. It's not related to ledger transport changes. Done.